src/program.cpp

```diff
@@ -808,12 +808,30 @@ double common_average(const std::vector<double>& v)
     return total / std::distance(v.begin() + n, v.end() - n);
 }
 
-std::string perf_group(const operation& op)
+std::string perf_group(instruction_ref ins, bool detailed)
 {
-    auto attr = op.attributes();
+    std::string result;
+    auto attr = ins->get_operator().attributes();
     if(attr.contains("group"))
-        return attr.at("group").to<std::string>();
-    return op.name();
+        result = attr.at("group").to<std::string>();
+    else
+        result = ins->get_operator().name();
+    if(detailed)
+    {
+        result += "<" + ins->get_shape().type_string();
```

Comment:
It is a bit misleading to use the output type as the input type across all the inputs.

Can we switch for sth like :  m x n x f32 for each input type instead and insert a `-> a x b x f32` at the end to indicate the output type ?

Reply:
(This is important for quant cases)

Reply:
Actually I think its better to keep it for now since it does give some information about the type. I can added abbreviated types for each tensor in a later PR.

Reply:
can you open an issue ?

Reply:
yes abbrev types should just work -- i dont consider additional 4 char are 'long' especially for a user who asks for a 'detailed' report